ctapipe/io/dl1eventsource.py

```diff
@@ -97,9 +97,23 @@ def __init__(self, input_url=None, config=None, parent=None, **kwargs):
 
         self.file_ = tables.open_file(self.input_url)
         self._full_subarray_info = SubarrayDescription.from_hdf(self.input_url)
+
         if self.allowed_tels:
```

Comment:
this code should go into a function or method of `EventSource` or probably even into `Subarray.select_subarray` itself.

You have it both here and in the simtel event source.